api/types/network/network.go

```diff
@@ -13,6 +13,13 @@ type IPAM struct {
 	Config  []IPAMConfig
 }
 
+// Options represents the network options for endpoint creation
+type Options struct {
```

Comment:
Options in a network api package can easily indicate options of the network object. In the context of the UX/API, I can see an argument to be made for `ServiceOption` under the network package. If you are okay, pls consider renaming it as `ServiceOption`.